packages/models-library/src/models_library/utils/database_models_factory.py

```diff
@@ -0,0 +1,94 @@
+""" Automatic creation of pydantic model classes from a sqlalchemy table
+
+SEE: Copied and adapted from https://github.com/tiangolo/pydantic-sqlalchemy/blob/master/pydantic_sqlalchemy/main.py
+"""
+
+import warnings
+from typing import Any, Container, Dict, Optional, Type
+from uuid import UUID
+
+import sqlalchemy as sa
+from pydantic import BaseConfig, BaseModel, Field, create_model
+from pydantic.types import NonNegativeInt
+
+warnings.warn(
+    "This is still a concept under development. "
+    "Currently only inteded for testing. "
+    "DO NOT USE in production.",
+    category=UserWarning,
+)
+
+
+class OrmConfig(BaseConfig):
+    orm_mode = True
+
+
+_RESERVED = {
+    "schema",
+    # e.g. Field name "schema" shadows a BaseModel attribute; use a different field name with "alias='schema'".
+}
+
+
+def sa_table_to_pydantic_model(
+    table: sa.Table,
+    *,
+    config: Type = OrmConfig,
+    exclude: Optional[Container[str]] = None,
+) -> Type[BaseModel]:
+
+    fields = {}
+    exclude = exclude or []
+
+    for column in table.columns:
+        name = str(column.name)
+
+        if name in exclude:
+            continue
+
+        field_args: Dict[str, Any] = {}
+
+        if name in _RESERVED:
+            field_args["alias"] = name
+            name = f"{table.name.lower()}_{name}"
+
+        pydantic_type: Optional[type] = None
```

Comment:
Is this meant to be a ref to the builtin func `type`?

Reply:
Here the annotation is that the variable ``pydantic_type`` will receive **any** type
AFAIK,  [``typing.Type``](https://docs.python.org/3/library/typing.html#typing.Type) is used as ``typing.Type[MyBaseClass]`` to express any type derived from ``MyBaseClass``, but not as a generic type